src/MessageList/MessageListItem.vala

```diff
@@ -203,33 +203,61 @@ public class Mail.MessageListItem : Gtk.ListBoxRow {
         };
         starred_button.get_style_context ().add_class (Gtk.STYLE_CLASS_FLAT);
 
-        var reply_item = new Gtk.MenuItem.with_label (_("Reply"));
-        reply_item.activate.connect (() => add_inline_composer (ComposerWidget.Type.REPLY));
+        var actions_menu_box = new Gtk.Box (VERTICAL, 0) {
+            margin_bottom = 3,
+            margin_top = 3
+        };
+
+        var actions_menu = new Gtk.Popover (null) {
+            child = actions_menu_box
+        };
+
+        var reply_item = new Gtk.ModelButton () {
+            label = _("Reply")
+        };
+        reply_item.clicked.connect (() => {
+            add_inline_composer (ComposerWidget.Type.REPLY);
+            actions_menu.popdown ();
```

Comment:
modelbuttons automatically pop down the menu so you don't need this :)

Reply:
In Gtk4 we'll use `Button` directly and style it with CSS and then yeah it'll be necessary to `popdown` then. So that's fair :+1:

Something I think we should do in a future branch is to maybe make a compose action that accepts the same arguments as `add_inline_composer` so that we can use actions everywhere and not have to have a bunch of different methods. Then we could use GLib.Menu here and save ourselves a lot of trouble :sweat_smile:

Reply:
Actually I've been working on this and have a somewhat working branch but dropped it as I thought it might get to complex, but I guess I'll start work on that again :)

Reply:
Cool! Also, I pinged you in Slack with some other thoughts/questions :)